src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -195,35 +201,55 @@ private RelationalStorage(string invariantName, string connectionString)
         }
 
 
-        private static async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, TResult> selector)
+        private async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, CancellationToken, Task<TResult>> selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
         {
-            using(var reader = await command.ExecuteReaderAsync().ConfigureAwait(continueOnCapturedContext: false))
-            {
-                return await SelectAsync(reader, selector).ConfigureAwait(false);
+            using(var reader = await command.ExecuteReaderAsync(commandBehavior, cancellationToken).ConfigureAwait(continueOnCapturedContext: false))
+            {                                                                
```

Comment:
the `MySQL` provider not being truly `async` is a bug. we can use `Task.Run` to mitigate the problem, and even implement cancellation by closing the connection on cancellation. anyway, I wouldn't want to treat this as a feature the provide lacks, it is a bug in the `MySql` provider.